src/main/java/seedu/sellsavvy/model/order/Date.java

```diff
@@ -33,24 +39,50 @@ public class Date {
      */
     public Date(String date) {
         requireNonNull(date);
-        checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidDateRegex(date), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidCalendarDate(date), MESSAGE_INVALID_DATE);
         value = date;
     }
 
     /**
-     * Returns if a given string is a valid date.
+     * Returns if a given string is of a valid date regex.
      */
-    public static boolean isValidDate(String test) {
+    public static boolean isValidDateRegex(String test) {
         if (!test.matches(REGEX_DATE)) {
             return false;
-        } else {
-            try {
-                LocalDate localDate = ParserUtil.parseLocalDate(test);
-            } catch (ParseException e) {
-                return false;
-            }
-            return true;
         }
+        return true;
+    }
+
+    /**
+     * Returns if a given string is a valid calendar date.
+     */
+    public static boolean isValidCalendarDate(String test) {
+        try {
+            LocalDate localDate = ParserUtil.parseLocalDate(test);
+        } catch (ParseException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Returns {@code Date} warning if order date has passed.
+     */
+    public String hasDateElapsed() {
```

Comment:
Given the method name is `hasDateElapsed`, wouldn't it make more sense if the method returns a boolean instead?
In EditOrderCommand, you can add `MESSAGE_INVALID_DATE` to the `feedbackToUser` variable if the date has already elapsed.

Reply:
Updated in commit.